src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/SelectionRows/Items/Items.tsx

```diff
@@ -26,52 +29,66 @@ const commonUploaderProps = {
   uiType: UploaderUiType.Secondary,
 };
 
-export const Items = ({ isSingle }: { isSingle?: boolean }) => {
+export const Items = ({ name, isSingle }: ItemsProps) => {
   const { t } = useTranslation('app');
 
-  const { watch, control, setValue } = useFormContext();
+  const { watch, control, setValue, getValues } = useFormContext();
 
-  const items = watch('selectionRows.items');
-  const options = watch('selectionRows.options');
-  const settings = watch('settings');
+  const optionsName = `${name}.responseValues.options`;
+  const rows = watch(`${name}.responseValues.rows`);
+  const options = watch(optionsName);
+  const settings = watch(`${name}.config`);
 
-  const hasTooltips = settings?.includes(ItemConfigurationSettings.HasTooltips);
-  const hasScores = settings?.includes(ItemConfigurationSettings.HasScores);
-  const hasRemoveButton = items?.length > 1;
+  const hasTooltips = get(settings, ItemConfigurationSettings.HasTooltips);
+  const hasScores = get(settings, ItemConfigurationSettings.HasScores);
+  const hasRemoveButton = rows?.length > 1;
 
-  const handleRemoveItem = (index: number) =>
+  const handleRemoveItem = (index: number) => {
     setValue(
-      'selectionRows.items',
-      items?.filter((item: SelectionRowsItem, key: number) => key !== index),
+      `${name}.responseValues.rows`,
+      rows?.filter((row: SingleAndMultipleSelectRow, key: number) => key !== index),
     );
 
-  return items?.map((item: SelectionRowsItem, index: number) => {
-    const name = `selectionRows.items[${index}]`;
+    if (hasScores) {
+      const dataMatrix = getValues(`${name}.responseValues.dataMatrix`);
```

Comment:
Can you move `${name}.responseValues.dataMatrix` in a variable dataMatrixName to reduce code duplication?

Reply:
done